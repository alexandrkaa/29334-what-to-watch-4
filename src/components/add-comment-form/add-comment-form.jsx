import React from 'react';
import PropTypes from 'prop-types';
import InputRadioStar from '../input-radio-star/input-radio-star.jsx';
import InputTextarea from '../input-textarea/input-textarea.jsx';
import {REVIEW_STARS_NUMBER} from '../../consts/consts.js';
import withReview from '../../hocs/with-comment/with-comment.js';
import Error from '../error/error.jsx';

const AddCommentForm = (props) => {
  const {
    onTextAreaChange,
    onRadioChange,
    comment,
    rating,
    onFormSubmit,
    postCommentInProgress,
    postCommentError,
    isFormValid,
  } = props;
  if (postCommentError) {
    return (
      <Error />
    );
  }


  return (
    <div className="add-review">
      <form onSubmit={onFormSubmit} className="add-review__form">
        <div className="rating">
          <div className="rating__stars">
            {
              new Array(REVIEW_STARS_NUMBER).fill(``).map((it, idx) => <InputRadioStar isDisabled={postCommentInProgress} key={idx} id={idx + 1} isChecked={rating === (idx + 1)} onRadioChange={onRadioChange} />)
            }
          </div>
        </div>

        <div className="add-review__text">
          <InputTextarea onTextAreaChange={onTextAreaChange} isDisabled={postCommentInProgress} comment={comment} />
          <div className="add-review__submit">
            <button className="add-review__btn" type="submit" disabled={!isFormValid || postCommentInProgress}>Post</button>
          </div>

        </div>
      </form>
    </div>
  );
};

AddCommentForm.propTypes = {
  onRadioChange: PropTypes.func.isRequired,
  onTextAreaChange: PropTypes.func.isRequired,
  comment: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  isFormValid: PropTypes.bool.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
  postCommentInProgress: PropTypes.bool.isRequired,
  postCommentError: PropTypes.bool.isRequired,
};

export {AddCommentForm};
export default withReview(AddCommentForm);
