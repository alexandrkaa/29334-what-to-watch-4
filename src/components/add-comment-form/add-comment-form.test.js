import React from 'react';
import renderer from 'react-test-renderer';
import {AddCommentForm} from './add-comment-form.jsx';

describe(`<AddCommentForm /> should match snapshot`, () => {
  const onRadioChange = jest.fn();
  const onTextAreaChange = jest.fn();
  const onFormSubmit = jest.fn();
  const comment = `Test comment`;
  const rating = 5;
  it(`<AddCommentForm /> should match snapshot without error`, () => {
    const tree = renderer
      .create(
          <AddCommentForm
            onRadioChange={onRadioChange}
            onTextAreaChange={onTextAreaChange}
            comment={comment}
            rating={rating}
            isFormValid={true}
            onFormSubmit={onFormSubmit}
            postCommentInProgress={false}
            postCommentError={false}
            history={{}}
            movieId={1}
            postComment={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`<AddCommentForm /> should match snapshot with error`, () => {
    const tree = renderer
      .create(
          <AddCommentForm
            onRadioChange={onRadioChange}
            onTextAreaChange={onTextAreaChange}
            comment={comment}
            rating={rating}
            isFormValid={true}
            onFormSubmit={onFormSubmit}
            postCommentInProgress={false}
            postCommentError={true}
            history={{}}
            movieId={1}
            postComment={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
