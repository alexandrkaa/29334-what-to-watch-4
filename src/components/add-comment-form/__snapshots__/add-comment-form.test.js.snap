// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCommentForm /> should match snapshot with error 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <h2>
    Sorry, error is occuried while loading content. Please try sometime later
  </h2>
  <svg
    enableBackground="new 0 0 100 100"
    height="125"
    viewBox="0 0 100 125"
    width="100"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <path
      d="M50 89c21.505 0 39-17.495 39-39S71.505 11 50 11 11 28.495 11 50s17.495 39 39 39zm0-75c19.85 0 36 16.15 36 36S69.85 86 50 86 14 69.85 14 50s16.15-36 36-36z"
    />
    <path
      d="M28.102 37.419a1.5 1.5 0 002.472 1.7c1-1.453 2.462-2.614 3.94-3.516 1.48.902 2.942 2.063 3.941 3.516a1.5 1.5 0 102.473-1.7c-.913-1.328-2.099-2.44-3.354-3.37 1.43-.6 2.48-.892 2.505-.9a1.5 1.5 0 00-.774-2.897c-.165.044-2.3.633-4.79 1.912-2.49-1.28-4.625-1.868-4.79-1.912a1.5 1.5 0 00-.774 2.898c.025.007 1.075.3 2.504.899-1.255.93-2.44 2.042-3.353 3.37zM59.072 37.419a1.5 1.5 0 002.473 1.7c1-1.453 2.462-2.614 3.94-3.516 1.48.902 2.941 2.063 3.94 3.516a1.5 1.5 0 102.473-1.7c-.913-1.328-2.098-2.44-3.353-3.37 1.43-.6 2.479-.892 2.505-.9a1.5 1.5 0 00-.775-2.897c-.165.044-2.3.633-4.79 1.912-2.49-1.28-4.625-1.868-4.79-1.912a1.5 1.5 0 00-.774 2.898c.026.007 1.076.3 2.505.899-1.255.93-2.44 2.042-3.354 3.37zM40.234 57.36a1.5 1.5 0 001.063 1.837c.052.014 2.525.689 5.62 2.134-2.411 1.602-4.752 3.622-6.469 6.118a1.5 1.5 0 002.47 1.7c1.795-2.608 4.457-4.657 7.087-6.202 2.624 1.543 5.28 3.592 7.076 6.202a1.496 1.496 0 002.086.386 1.5 1.5 0 00.385-2.086c-1.716-2.494-4.055-4.514-6.465-6.115 3.094-1.445 5.565-2.123 5.617-2.137a1.5 1.5 0 00-.774-2.899c-.268.072-3.853 1.053-7.93 3.207-4.077-2.154-7.662-3.135-7.93-3.207a1.497 1.497 0 00-1.836 1.063z"
    />
  </svg>
</div>
`;

exports[`<AddCommentForm /> should match snapshot without error 1`] = `
<div
  className="add-review"
>
  <form
    className="add-review__form"
    onSubmit={[MockFunction]}
  >
    <div
      className="rating"
    >
      <div
        className="rating__stars"
      >
        <input
          checked={false}
          className="rating__input"
          disabled={false}
          id="star-1"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <label
          className="rating__label"
          htmlFor="star-1"
        >
          Rating 1
        </label>
        <input
          checked={false}
          className="rating__input"
          disabled={false}
          id="star-2"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          className="rating__label"
          htmlFor="star-2"
        >
          Rating 2
        </label>
        <input
          checked={false}
          className="rating__input"
          disabled={false}
          id="star-3"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          className="rating__label"
          htmlFor="star-3"
        >
          Rating 3
        </label>
        <input
          checked={false}
          className="rating__input"
          disabled={false}
          id="star-4"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <label
          className="rating__label"
          htmlFor="star-4"
        >
          Rating 4
        </label>
        <input
          checked={true}
          className="rating__input"
          disabled={false}
          id="star-5"
          name="rating"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <label
          className="rating__label"
          htmlFor="star-5"
        >
          Rating 5
        </label>
      </div>
    </div>
    <div
      className="add-review__text"
    >
      <textarea
        className="add-review__textarea"
        disabled={false}
        id="review-text"
        name="review-text"
        onChange={[Function]}
        placeholder="Review text"
        value="Test comment"
      />
      <div
        className="add-review__submit"
      >
        <button
          className="add-review__btn"
          disabled={false}
          type="submit"
        >
          Post
        </button>
      </div>
    </div>
  </form>
</div>
`;
